Source/Core/MapProjection.js

```diff
@@ -65,5 +91,29 @@ define([
      */
     MapProjection.prototype.unproject = DeveloperError.throwInstantiationError;
 
+    var maxcoordRectangleScratch = new Rectangle();
+    var rectangleCenterScratch = new Cartographic();
+    /**
+     * Approximates the extents of a map projection in 2D.
+     *
+     * @function
+     *
+     * @param {MapProjection} mapProjection A map projection from cartographic coordinates to 2D space.
+     * @param {Cartesian3} [result] Optional result parameter.
+     * @private
+     */
+    MapProjection.approximateMaximumCoordinate = function(mapProjection, result) {
+        Check.defined('mapProjection', mapProjection);
+
+        var maxCoord = defaultValue(result, new Cartesian3());
+        var projectedExtents = Rectangle.approximateProjectedExtents(Rectangle.MAX_VALUE, mapProjection, maxcoordRectangleScratch);
+        var projectedCenter = Rectangle.center(projectedExtents, rectangleCenterScratch);
+
+        maxCoord.x = projectedCenter.longitude + projectedExtents.width * 0.5;
+        maxCoord.y = projectedCenter.latitude + projectedExtents.height * 0.5;
+
```

Comment:
z is never set, that seems like a bug?

Reply:
None of the callees do anything with the z value. This function is just here because all the callees used to assume a symmetrical rectangular projection when setting up params like how far away from center the camera can go, so they'd just project a corner of Rectangle.MAX_VALUE. This was causing bizarre camera behavior for projections where projecting a corner from Rectangle.MAX_VALUE just 

I'll change this to return a `Cartesian2` instead and update the doc.